lib/streamlit/script_runner.py

```diff
@@ -340,13 +341,10 @@ def _run_script(self, rerun_data):
                     # The old states, used to skip callbacks if values
                     # haven't changed, are also preserved in the
                     # WidgetManager.
-                    self._widget_mgr.mark_widgets_as_old()
-                    self._widget_mgr.set_widget_states(rerun_data.widget_states)
+                    self._session_state._compact_state()
```

Comment:
Is this state compaction necessary since one is also done in `self.session_state._call_callbacks`?

Reply:
This one should be, but I think the other one isn't necessary once I change the priority order. I was doing both to clear out space for this batch of widget states, then move those in so that the session state values set in callbacks would show up over those widget states. But now I think the latter is unneeded.

Reply:
Seemingly removing the other causes a test to fail, and I am too tired to want to investigate why today